core/templates/dev/head/domain/exploration/QuestionObjectFactory.js

```diff
@@ -79,6 +80,18 @@ oppia.factory('QuestionObjectFactory', [function() {
     return this._answerGroups.length > 0;
   };
 
+  Question.prototype.getId = function () {
+    return this.getStateName().toLowerCase().split(' ').join('');
```

Comment:
Not sure -- you'll need to check. I think there's a constant somewhere denoting what the allowed chars are, it would be good to add a karma test to ensure that your delimiter of choice doesn't pass state name validation (and add a comment explaining why this test must not be modified).